src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/ScannedTemplateInfo.cs

```diff
@@ -0,0 +1,124 @@
+    /// Before running the template the caller should select localization and host file to use.
+    /// Use <see cref="RunnableProjectConfig"/> to load template for running.
+    /// </summary>
+    internal class ScannedTemplateInfo : DirectoryBasedTemplate, IScanTemplateInfo
+    {
+        internal const string HostTemplateFileConfigBaseName = ".host.json";
+        internal const string LocalizationFilePrefix = "templatestrings.";
+        internal const string LocalizationFileExtension = ".json";
+
+        private readonly IReadOnlyDictionary<string, IFile> _hostConfigFiles;
+        private readonly IReadOnlyDictionary<string, TemplateLocalization> _localizations;
+
+        /// <summary>
+        /// Creates instance of the class based on configuration from <paramref name="templateFile"/>.
+        /// </summary>
+        public ScannedTemplateInfo(IEngineEnvironmentSettings settings, IGenerator generator, IFile templateFile) : base(settings, generator, templateFile)
+        {
+            _hostConfigFiles = FindHostTemplateConfigFiles();
+            _localizations = FindLocalizations();
+        }
+
+        public IReadOnlyDictionary<string, ILocalizationLocator> Localizations => _localizations.ToDictionary(l => l.Key, l => (ILocalizationLocator)l.Value);
+
+        public IReadOnlyDictionary<string, string> HostConfigFiles => _hostConfigFiles.ToDictionary(f => f.Key, f => f.Value.FullPath);
+
+        public override IFile ConfigFile => base.ConfigFile ?? throw new InvalidOperationException();
+
+        public override IDirectory ConfigDirectory => base.ConfigDirectory ?? throw new InvalidOperationException();
+
+        /// <summary>
+        /// Attempts to find the host configuration files.
+        /// </summary>
+        private IReadOnlyDictionary<string, IFile> FindHostTemplateConfigFiles()
+        {
+            Dictionary<string, IFile> allHostFilesForTemplate = new();
+
+            foreach (IFile hostFile in ConfigDirectory.EnumerateFiles($"*{HostTemplateFileConfigBaseName}", SearchOption.TopDirectoryOnly))
+            {
+                allHostFilesForTemplate.Add(hostFile.Name.Replace(HostTemplateFileConfigBaseName, string.Empty), hostFile);
+                Logger.LogDebug($"Found *{HostTemplateFileConfigBaseName} at {hostFile.GetDisplayPath()}.");
+            }
+
+            return allHostFilesForTemplate;
+        }
+
+        private IReadOnlyDictionary<string, TemplateLocalization> FindLocalizations()
+        {
+            Dictionary<string, TemplateLocalization> localizations = new();
+            IDirectory? localizeFolder = ConfigDirectory.DirectoryInfo("localize");
+            if (localizeFolder == null || !localizeFolder.Exists)
+            {
+                return localizations;
+            }
+
+            foreach (IFile locFile in localizeFolder.EnumerateFiles(LocalizationFilePrefix + "*" + LocalizationFileExtension, SearchOption.AllDirectories))
+            {
```

Comment:
We should validate that the localization tokens in filenames are representing valid `CultureInfo` - to prevent situation, where someone accidentaly creates invalid localization file (generating files through tooling should avoid this - but there is still chance that somene creates the files manually).

Reply:
added